src/realm/list.cpp

```diff
@@ -192,6 +193,77 @@ void Lst<T>::distinct(std::vector<size_t>& indices, util::Optional<bool> sort_or
     }
 }
 
+/***************************** Lst<Stringdata> ******************************/
+
+template <>
+void Lst<StringData>::do_insert(size_t ndx, StringData value)
+{
+    if (auto index = m_obj.get_table()->get_search_index(m_col_key)) {
+        if (m_tree->find_first(value) == realm::not_found) {
```

Comment:
I suppose the idea behind handling duplicates at this level is to optimize the find operation. But the cost of O(N) on insert/set/remove seems very high. What would you think about allowing duplicates into the index to make these operations faster and less error prone? Then we would have a small sort/unique cost applied in `StringIndex::from_list_all` to remove duplicates from the results. Or perhaps we could even return duplicates there? Then users could see how many items in the list match which could be a useful feature?

Reply:
I'll give it a try.

Reply:
I am a bit wary of changing the invariant that there are no duplicate object keys in the index. I have not changed the code so that inserting a value twice is idempotent, so at least we can avoid the O(N) behavior on insertions. Would that be an acceptable compromise?

Reply:
I'm good with that. Thanks!